apps/src/tutorialExplorer/tutorialExplorer.js

```diff
@@ -54,29 +66,205 @@ const TutorialExplorer = React.createClass({
       const itemIndex = this.state.filters[filterGroup].indexOf(filterEntry);
       newState = state.updateIn(['filters', filterGroup], arr => arr.splice(itemIndex, 1));
     }
-    this.setState(newState.toJS());
+
+    this.setState(newState.toJS(), function () {
+      const { filteredTutorials, filteredTutorialsForLocale } = this.filterTutorials(this.state.filters);
+      this.setState({
+        filteredTutorials: filteredTutorials,
+        filteredTutorialsCount: filteredTutorials.length,
+        filteredTutorialsForLocale: filteredTutorialsForLocale
+      });
+    });
```

Comment:
Is that true? `state` is the Immutable version of `this.state` (and thus should represent the entirety of the state). `newState` is then the Immutable version of `this.state` with your `updateIn` step applied. Doesn't that mean it should represent the entirety of the resulting state? (and be identical to this.state inside your callback?)